### ISSUE TITLE
Clarification with training procedure for learning rate and unexpected keys

### ISSUE THREAD

**Issue by ashj9** [DevashishPrasad/CascadeTabNet#83]
First of all, great work guys!!
I am planning to train the network on my dateset. However, before that I am trying to reproduce the training procedure to get comfortable with. I have started training on general dataset mentioned in the paper. However, I have doubts on two of the logs that are generated.

After first 50 iterations, it says the learning rate after 50 iterations is 0.00048. Is this expected value?

```
2020-09-15 22:48:51,500 - mmdet - INFO - workflow: [('train', 1)], max: 50 epochs
2020-09-15 22:51:34,065 - mmdet - INFO - Epoch [1][50/966]	lr: 0.00048, eta: 1 day, 19:34:13, time: 3.251, data_time: 0.016, memory: 9193, loss_rpn_cls: 0.3312, loss_rpn_bbox: 0.0207, s0.loss_cls: 0.7532, s0.acc: 94.5762, s0.loss_bbox: 0.0319, s0.loss_mask: 0.5419, s1.loss_cls: 0.4374, s1.acc: 92.7012, s1.loss_bbox: 0.0097, s1.loss_mask: 0.1658, s2.loss_cls: 0.2705, s2.acc: 87.8438, s2.loss_bbox: 0.0014, s2.loss_mask: 0.0660, loss: 2.6297
```

Secondly,  
```
2020-09-15 22:48:48,617 - mmdet - INFO - load model from: open-mmlab://msra/hrnetv2_w32
2020-09-15 22:48:49,162 - mmdet - WARNING - The model and loaded state dict do not match exactly

unexpected key in source state_dict: incre_modules.0.0.conv1.weight, incre_modules.0.0.bn1.weight, 
incre_modules.0.0.bn1.bias, incre_modules.0.0.bn1.running_mean, incre_modules.0.0.bn1.running_var, 
incre_modules.0.0.bn1.num_batches_tracked, incre_modules.0.0.conv2.weight, incre_modules.0.0.bn2.weight, 
incre_modules.0.0.bn2.bias, incre_modules.0.0.bn2.running_mean, incre_modules.0.0.bn2.running_var, 
incre_modules.0.0.bn2.num_batches_tracked, incre_modules.0.0.conv3.weight, incre_modules.0.0.bn3.weight, 
incre_modules.0.0.bn3.bias, incre_modules.0.0.bn3.running_mean, incre_modules.0.0.bn3.running_var, 
incre_modules.0.0.bn3.num_batches_tracked, incre_modules.0.0.downsample.0.weight, incre_modules.0.0.downsample.1.weight, incre_modules.0.0.downsample.1.bias, 
incre_modules.0.0.downsample.1.running_mean, incre_modules.0.0.downsample.1.running_var, 
incre_modules.0.0.downsample.1.num_batches_tracked, incre_modules.1.0.conv1.weight, incre_modules.1.0.bn1.weight, incre_modules.1.0.bn1.bias, incre_modules.1.0.bn1.running_mean, incre_modules.1.0.bn1.running_var, 
incre_modules.1.0.bn1.num_batches_tracked, incre_modules.1.0.conv2.weight, incre_modules.1.0.bn2.weight, 
incre_modules.1.0.bn2.bias, incre_modules.1.0.bn2.running_mean, incre_modules.1.0.bn2.running_var, 
incre_modules.1.0.bn2.num_batches_tracked, incre_modules.1.0.conv3.weight, incre_modules.1.0.bn3.weight, 
incre_modules.1.0.bn3.bias, incre_modules.1.0.bn3.running_mean, incre_modules.1.0.bn3.running_var, 
incre_modules.1.0.bn3.num_batches_tracked, incre_modules.1.0.downsample.0.weight, 
incre_modules.1.0.downsample.1.weight, incre_modules.1.0.downsample.1.bias, 
incre_modules.1.0.downsample.1.running_mean, incre_modules.1.0.downsample.1.running_var, 
incre_modules.1.0.downsample.1.num_batches_tracked, incre_modules.2.0.conv1.weight, incre_modules.2.0.bn1.weight, 
incre_modules.2.0.bn1.bias, incre_modules.2.0.bn1.running_mean, incre_modules.2.0.bn1.running_var, 
incre_modules.2.0.bn1.num_batches_tracked, incre_modules.2.0.conv2.weight, incre_modules.2.0.bn2.weight, 
incre_modules.2.0.bn2.bias, incre_modules.2.0.bn2.running_mean, incre_modules.2.0.bn2.running_var, 
incre_modules.2.0.bn2.num_batches_tracked, incre_modules.2.0.conv3.weight, incre_modules.2.0.bn3.weight, 
incre_modules.2.0.bn3.bias, incre_modules.2.0.bn3.running_mean, incre_modules.2.0.bn3.running_var, 
incre_modules.2.0.bn3.num_batches_tracked, incre_modules.2.0.downsample.0.weight, 
incre_modules.2.0.downsample.1.weight, incre_modules.2.0.downsample.1.bias, 
incre_modules.2.0.downsample.1.running_mean, incre_modules.2.0.downsample.1.running_var, 
incre_modules.2.0.downsample.1.num_batches_tracked, incre_modules.3.0.conv1.weight, incre_modules.3.0.bn1.weight, incre_modules.3.0.bn1.bias, incre_modules.3.0.bn1.running_mean, incre_modules.3.0.bn1.running_var, 
incre_modules.3.0.bn1.num_batches_tracked, incre_modules.3.0.conv2.weight, incre_modules.3.0.bn2.weight, 
incre_modules.3.0.bn2.bias, incre_modules.3.0.bn2.running_mean, incre_modules.3.0.bn2.running_var, 
incre_modules.3.0.bn2.num_batches_tracked, incre_modules.3.0.conv3.weight, incre_modules.3.0.bn3.weight, 
incre_modules.3.0.bn3.bias, incre_modules.3.0.bn3.running_mean, incre_modules.3.0.bn3.running_var, 
incre_modules.3.0.bn3.num_batches_tracked, incre_modules.3.0.downsample.0.weight, 
incre_modules.3.0.downsample.1.weight, incre_modules.3.0.downsample.1.bias, 
incre_modules.3.0.downsample.1.running_mean, incre_modules.3.0.downsample.1.running_var, 
incre_modules.3.0.downsample.1.num_batches_tracked, downsamp_modules.0.0.weight, downsamp_modules.0.0.bias, downsamp_modules.0.1.weight, downsamp_modules.0.1.bias, downsamp_modules.0.1.running_mean, downsamp_modules.0.1.running_var, downsamp_modules.0.1.num_batches_tracked, downsamp_modules.1.0.weight, downsamp_modules.1.0.bias, downsamp_modules.1.1.weight, downsamp_modules.1.1.bias, 
downsamp_modules.1.1.running_mean, downsamp_modules.1.1.running_var, 
downsamp_modules.1.1.num_batches_tracked, downsamp_modules.2.0.weight, downsamp_modules.2.0.bias, 
downsamp_modules.2.1.weight, downsamp_modules.2.1.bias, downsamp_modules.2.1.running_mean, 
downsamp_modules.2.1.running_var, downsamp_modules.2.1.num_batches_tracked, final_layer.0.weight, 
final_layer.0.bias, final_layer.1.weight, final_layer.1.bias, final_layer.1.running_mean, final_layer.1.running_var, 
final_layer.1.num_batches_tracked, classifier.weight, classifier.bias

```

The message says, " mmdet - WARNING - The model and loaded state dict do not match exactly" followed by some keys that corresponds to some layers.

Could you also please help me confirming if this is also an expected behaviour? Or I need to correct my training procedure?

**Comment by iazdan** [comment]
> First of all, great work guys!!
> I am planning to train the network on my dateset. However, before that I am trying to reproduce the training procedure to get comfortable with. I have started training on general dataset mentioned in the paper. However, I have doubts on two of the logs that are generated.
> 
> After first 50 iterations, it says the learning rate after 50 iterations is 0.00048. Is this expected value?
> 
> ```
> 2020-09-15 22:48:51,500 - mmdet - INFO - workflow: [('train', 1)], max: 50 epochs
> 2020-09-15 22:51:34,065 - mmdet - INFO - Epoch [1][50/966]	lr: 0.00048, eta: 1 day, 19:34:13, time: 3.251, data_time: 0.016, memory: 9193, loss_rpn_cls: 0.3312, loss_rpn_bbox: 0.0207, s0.loss_cls: 0.7532, s0.acc: 94.5762, s0.loss_bbox: 0.0319, s0.loss_mask: 0.5419, s1.loss_cls: 0.4374, s1.acc: 92.7012, s1.loss_bbox: 0.0097, s1.loss_mask: 0.1658, s2.loss_cls: 0.2705, s2.acc: 87.8438, s2.loss_bbox: 0.0014, s2.loss_mask: 0.0660, loss: 2.6297
> ```
> 
> Secondly,
> 
> ```
> 2020-09-15 22:48:48,617 - mmdet - INFO - load model from: open-mmlab://msra/hrnetv2_w32
> 2020-09-15 22:48:49,162 - mmdet - WARNING - The model and loaded state dict do not match exactly
> 
> unexpected key in source state_dict: incre_modules.0.0.conv1.weight, incre_modules.0.0.bn1.weight, 
> incre_modules.0.0.bn1.bias, incre_modules.0.0.bn1.running_mean, incre_modules.0.0.bn1.running_var, 
> incre_modules.0.0.bn1.num_batches_tracked, incre_modules.0.0.conv2.weight, incre_modules.0.0.bn2.weight, 
> incre_modules.0.0.bn2.bias, incre_modules.0.0.bn2.running_mean, incre_modules.0.0.bn2.running_var, 
> incre_modules.0.0.bn2.num_batches_tracked, incre_modules.0.0.conv3.weight, incre_modules.0.0.bn3.weight, 
> incre_modules.0.0.bn3.bias, incre_modules.0.0.bn3.running_mean, incre_modules.0.0.bn3.running_var, 
> incre_modules.0.0.bn3.num_batches_tracked, incre_modules.0.0.downsample.0.weight, incre_modules.0.0.downsample.1.weight, incre_modules.0.0.downsample.1.bias, 
> incre_modules.0.0.downsample.1.running_mean, incre_modules.0.0.downsample.1.running_var, 
> incre_modules.0.0.downsample.1.num_batches_tracked, incre_modules.1.0.conv1.weight, incre_modules.1.0.bn1.weight, incre_modules.1.0.bn1.bias, incre_modules.1.0.bn1.running_mean, incre_modules.1.0.bn1.running_var, 
> incre_modules.1.0.bn1.num_batches_tracked, incre_modules.1.0.conv2.weight, incre_modules.1.0.bn2.weight, 
> incre_modules.1.0.bn2.bias, incre_modules.1.0.bn2.running_mean, incre_modules.1.0.bn2.running_var, 
> incre_modules.1.0.bn2.num_batches_tracked, incre_modules.1.0.conv3.weight, incre_modules.1.0.bn3.weight, 
> incre_modules.1.0.bn3.bias, incre_modules.1.0.bn3.running_mean, incre_modules.1.0.bn3.running_var, 
> incre_modules.1.0.bn3.num_batches_tracked, incre_modules.1.0.downsample.0.weight, 
> incre_modules.1.0.downsample.1.weight, incre_modules.1.0.downsample.1.bias, 
> incre_modules.1.0.downsample.1.running_mean, incre_modules.1.0.downsample.1.running_var, 
> incre_modules.1.0.downsample.1.num_batches_tracked, incre_modules.2.0.conv1.weight, incre_modules.2.0.bn1.weight, 
> incre_modules.2.0.bn1.bias, incre_modules.2.0.bn1.running_mean, incre_modules.2.0.bn1.running_var, 
> incre_modules.2.0.bn1.num_batches_tracked, incre_modules.2.0.conv2.weight, incre_modules.2.0.bn2.weight, 
> incre_modules.2.0.bn2.bias, incre_modules.2.0.bn2.running_mean, incre_modules.2.0.bn2.running_var, 
> incre_modules.2.0.bn2.num_batches_tracked, incre_modules.2.0.conv3.weight, incre_modules.2.0.bn3.weight, 
> incre_modules.2.0.bn3.bias, incre_modules.2.0.bn3.running_mean, incre_modules.2.0.bn3.running_var, 
> incre_modules.2.0.bn3.num_batches_tracked, incre_modules.2.0.downsample.0.weight, 
> incre_modules.2.0.downsample.1.weight, incre_modules.2.0.downsample.1.bias, 
> incre_modules.2.0.downsample.1.running_mean, incre_modules.2.0.downsample.1.running_var, 
> incre_modules.2.0.downsample.1.num_batches_tracked, incre_modules.3.0.conv1.weight, incre_modules.3.0.bn1.weight, incre_modules.3.0.bn1.bias, incre_modules.3.0.bn1.running_mean, incre_modules.3.0.bn1.running_var, 
> incre_modules.3.0.bn1.num_batches_tracked, incre_modules.3.0.conv2.weight, incre_modules.3.0.bn2.weight, 
> incre_modules.3.0.bn2.bias, incre_modules.3.0.bn2.running_mean, incre_modules.3.0.bn2.running_var, 
> incre_modules.3.0.bn2.num_batches_tracked, incre_modules.3.0.conv3.weight, incre_modules.3.0.bn3.weight, 
> incre_modules.3.0.bn3.bias, incre_modules.3.0.bn3.running_mean, incre_modules.3.0.bn3.running_var, 
> incre_modules.3.0.bn3.num_batches_tracked, incre_modules.3.0.downsample.0.weight, 
> incre_modules.3.0.downsample.1.weight, incre_modules.3.0.downsample.1.bias, 
> incre_modules.3.0.downsample.1.running_mean, incre_modules.3.0.downsample.1.running_var, 
> incre_modules.3.0.downsample.1.num_batches_tracked, downsamp_modules.0.0.weight, downsamp_modules.0.0.bias, downsamp_modules.0.1.weight, downsamp_modules.0.1.bias, downsamp_modules.0.1.running_mean, downsamp_modules.0.1.running_var, downsamp_modules.0.1.num_batches_tracked, downsamp_modules.1.0.weight, downsamp_modules.1.0.bias, downsamp_modules.1.1.weight, downsamp_modules.1.1.bias, 
> downsamp_modules.1.1.running_mean, downsamp_modules.1.1.running_var, 
> downsamp_modules.1.1.num_batches_tracked, downsamp_modules.2.0.weight, downsamp_modules.2.0.bias, 
> downsamp_modules.2.1.weight, downsamp_modules.2.1.bias, downsamp_modules.2.1.running_mean, 
> downsamp_modules.2.1.running_var, downsamp_modules.2.1.num_batches_tracked, final_layer.0.weight, 
> final_layer.0.bias, final_layer.1.weight, final_layer.1.bias, final_layer.1.running_mean, final_layer.1.running_var, 
> final_layer.1.num_batches_tracked, classifier.weight, classifier.bias
> ```
> 
> The message says, " mmdet - WARNING - The model and loaded state dict do not match exactly" followed by some keys that corresponds to some layers.
> 
> Could you also please help me confirming if this is also an expected behaviour? Or I need to correct my training procedure?

Were you able to fix this issue?
